packages/host/app/components/editor/go.gts

```diff
@@ -219,20 +221,70 @@ export default class Go extends Component<Signature> {
     return file.write(content);
   }
 
-  async saveSingleCardDocument(json: any) {
+  async saveJSONInstance(json: any) {
     let realmPath = new RealmPaths(this.cardService.defaultURL);
-    let url = realmPath.fileURL(
+    let cardUrl = realmPath.fileURL(
       this.args.openFiles.path!.replace(/\.json$/, ''),
     );
 
     try {
-      await this.cardService.saveCardDocument(json, url);
-      await this.loadCard.perform(url);
+      await this.postOrPatchCardDocument(json, cardUrl);
+      await this.loadCard.perform(cardUrl);
     } catch (e) {
       console.log('Failed to save single card document', e);
     }
   }
 
+  // NOTE: do not use this function outside of this component unless you are serializing a file
+  // this has been copied from runtime/realm.ts
+  private async fileSerialization(
```

Comment:
I think you did this exactly backwards from what I asked. I was describing that files are serializated in a very special way. But I was requesting that you _reverse_ this serialization as soon as possible so that outside of whatever component first encounters the file serialization (outside of faithfully representing this in Monaco), so that to the rest of the system it looks like the wire serialization that the rest of the system handles. Like I was saying before it’s the realm servers job to serialize this into the file serialization (which it does in thr create and patch card realm server functions). The host should not never to do this.

Reply:
I’m the file serialization you can see that it mainly just removes the `id` property, the `included` property and the removes the `data` property from each relationship. 

To reverse this you would _add_ the id property to the document (the url of the card without .json extension). 
For the `included`property you don’t have to add it back when reversing, the `included` is optional for the wire format, it’s used to tell the client info about linked cards that that have a relationship to the resource in the `data` field so that the client doesn’t have to do the extra fetch. (And coincidentally we found out just this week that for some reason AWS infra is returning 403 for POST requests that have this property—super weird)
For the resource `data` fields that is also optional in the wire format. The card runtime was written specifically to use the sibling `self` field instead of `data` field when processing link relationships. This was a decision we made at the time so I think it’s ok to omit these. If I’m the future we ever do need to use the relationship `data` field we might have to adjust this reversal to add them in. 

So ultimately to reverse the file serialization it’s just adding the id back in. But that important as the id is an indicator in the doc whether the card is saved or not.